### PR TITLE
HCM Project Onboarding Workflow

### PR DESCRIPTION
**Pull request by gaelyn**

### CODE REVIEW

File: src/shared/eventHandlers/hcmPrefillData.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+              employee_id: { value: id },
+              name: { value: name },
+              email: { value: email },
+              department_id: { value: departmentId },
+            })
+          );
+
+          await api.records.insert(employeesSheetId, mappedEmployees);
+        }
+      } catch (error) {
```
Comment (johnmosesman) [inline src/shared/eventHandlers/hcmPrefillData.ts:85]: Do we prefill employees in HCM? I thought we only validated against existing IDs

File: src/shared/hcm-show-api-service.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+//   phoneNumber: string;
+//   departmentId: string;
+//   jobId: string;
+// };
+
+// type JobResult = {
+//   externalJobId: string;
+//   name: string;
+//   description: string;
+// };
```
Comment (johnmosesman) [inline src/shared/hcm-show-api-service.ts:25]: Typing the API responses with a `as Jobresult etc` can help in the functions they're called

**Review by johnmosesman — COMMENTED** [review]
Couple comments but looking good to me!

File: src/shared/hcm-show-api-service.ts (hunk 2 above)
Comment (gaelyn) [reply src/shared/hcm-show-api-service.ts]: Ah thanks, meant to add them back in

File: src/shared/eventHandlers/hcmPrefillData.ts (hunk 1 above)
Comment (gaelyn) [reply src/shared/eventHandlers/hcmPrefillData.ts]: You're right, not sure what I was thinking

File: src/shared/externalContraints/externalConstraints.ts [hunk 3]
```diff
@@ -1,55 +1,56 @@
...
 
-const phoneUtil = PhoneNumberUtil.getInstance()
+const phoneUtil = PhoneNumberUtil.getInstance();
 
 // Function to format a date string based on the specified format
 function formatDate(dateString, outputFormat) {
   const inputFormats = [
     'yyyy-MM-dd',
     'MM/dd/yyyy',
+    'MM/dd/yy',
```
Comment (gaelyn) [inline src/shared/externalContraints/externalConstraints.ts:11]: this is the only actual diff here, the rest is formatting.

File: src/shared/validations/validations.ts [hunk 4]
```diff
@@ -2,14 +2,35 @@
...
 
 /**
  * @param {FlatfileListener} listener - The FlatfileListener instance to which custom
  * validations are attached. This object acts as the conduit through which records are
  * received from Flatfile, validated, and potentially modified or flagged for correction.
  */
 export function validations(listener: FlatfileListener) {
-  listener.use(plmValidations)
+  listener.use(plmValidations);
+  listener.use(hcmValidations);
```
Comment (johnmosesman) [inline src/shared/validations/validations.ts:16]: Can we split these validations into project specific files? Running plm validations against hcm or vice versa is confusing and also not needed

**Review by johnmosesman — APPROVED** [review]
One comment which can be address in another PR otherwise +1!

File: src/shared/validations/validations.ts (hunk 4 above)
Comment (gaelyn) [reply src/shared/validations/validations.ts:16]: yeah that makes sense

**Pull request merged**